vm/systemSmartContracts/delegation.go

```diff
@@ -661,7 +673,7 @@ func (d *delegation) unBondNodes(args *vmcommon.ContractCallInput) vmcommon.Retu
 		return vmcommon.UserError
 	}
 
-	err = verifyIfBLSPubKeysExist(dStatus.UnStakedKeys, args.Arguments)
+	err = verifyIfBLSPubKeysExist(dStatus.StakedKeys, args.Arguments)
```

Comment:
Maybe you can add this check also for  NotStakedKeys

Reply:
done. changed.